### PR TITLE
MWDDS-238 Add v4 swagger specification to activation ms

### PR DESCRIPTION
**Pull request by oparashchyn**
TMF Service activation API v4 has been added.
Both v2 and v4 version are supported

### CODE REVIEW

File: src/main/java/com/icthh/xm/tmf/ms/activation/service/SagaTransactionService.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.web.context.request.RequestAttributes;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class SagaTransactionService {
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/tmf/ms/activation/service/SagaTransactionService.java:19]: for me better to name factory or mapper or smth like this
we have SagaTransactionService it's about saga transaction logic

Comment (oparashchyn) [reply src/main/java/com/icthh/xm/tmf/ms/activation/service/SagaTransactionService.java]: I have renamed SagaTransactionService to SagaTransactionFactory.
By the way, there is no another SagaTransactionService class, SagaService contains all saga transaction logic

File: build.gradle [hunk 2]
```diff
@@ -374,6 +374,8 @@ dependencies {
     testImplementation "org.testcontainers:postgresql:1.6.0"
     testImplementation "com.h2database:h2"
     liquibaseRuntime "com.h2database:h2"
+    implementation "com.oracle.ojdbc:ojdbc8"
+    liquibaseRuntime "com.oracle.ojdbc:ojdbc8"
```
Comment (vkirichenko) [inline build.gradle:378]: why oracle dependency is added directly? In another ms - this dependency included dynamically on docker level.

Comment (oparashchyn) [reply build.gradle:378]: Ok, I will remove.
But at the same time other microservices have this dependency in gradle, for example loyalty.

Comment (ishkurko) [reply build.gradle]: Better remove from another microservice also. Oracle JDBC driver requires additional licensing and not applied with A2.0 license.

Comment (oparashchyn) [reply build.gradle]: Ok, understood

**Pull request merged**